src/components/templates/product-availability-modal/available-channels-modal-screen.tsx

```diff
@@ -0,0 +1,99 @@
+import { SalesChannel } from "@medusajs/medusa"
+import React, { useState } from "react"
+import { usePagination, useRowSelect, useTable } from "react-table"
+import SalesChannelAvailabilityTable, {
+  SalesChannelTableActions,
+  useAvailableChannelsModalTableColumns,
+} from "./sales-channel-availability-table"
+
+type AvailableChannelsModalScreenProps = {
+  salesChannels: SalesChannel[]
+  setSelectedSalesChannels: (salesChannels: SalesChannel[]) => void
+}
+
+const AvailableChannelsModalScreen: React.FC<AvailableChannelsModalScreenProps> = ({
+  salesChannels,
+  setSelectedSalesChannels,
+}) => {
+  const limit = 15
+  const numPages = Math.ceil(salesChannels?.length / limit)
+
+  const [offset, setOffset] = useState(0)
+  const [selectedRowIds, setSelectedRowIds] = useState<string[]>([])
+  const [query, setQuery] = useState("")
+
+  const [columns] = useAvailableChannelsModalTableColumns()
+
+  const tableState = useTable(
+    {
+      columns,
+      data: filterSalesChannels(salesChannels),
+      manualPagination: true,
+      initialState: {
+        pageIndex: Math.floor(offset / limit),
+        pageSize: limit,
+      },
+      autoResetPage: false,
+      autoResetSelectedRows: false,
+      getRowId: (row) => row.id,
+      pageCount: numPages,
+    },
+    usePagination,
+    useRowSelect
+  )
+
+  function filterSalesChannels(channels: SalesChannel[]) {
+    if (!query) {
+      return channels
+    }
+
+    return channels.filter((ch) => {
+      const filter = query.toLowerCase()
+      return (
+        !!ch.name.toLowerCase().match(filter) ||
+        !!ch.description?.toLowerCase().match(filter)
+      )
+    })
+  }
+
+  const onDeselect = () => {
+    setSelectedRowIds([])
+    tableState.toggleAllRowsSelected(false)
+  }
+
+  const onAddSalesChannelsToSelected = (selectedSalesChannels) => {
+    setSelectedSalesChannels([...salesChannels, ...selectedSalesChannels])
+    tableState.toggleAllRowsSelected(false)
+  }
+
+  const onRemove = () => {
+    const remainingSalesChannels = salesChannels.filter(
+      (ch) => !selectedRowIds.includes(ch.id)
+    )
+    setSelectedSalesChannels(remainingSalesChannels)
+    onDeselect()
+  }
+
+  return (
+    <SalesChannelAvailabilityTable
+      tableAction={
+        <SalesChannelTableActions
+          selectedRowIds={selectedRowIds}
+          chosenSalesChannelIds={salesChannels.map((sc) => sc.id)}
```

Comment:
suggestion: `selectedSalesChannels` to keep consistency as of now it is called chosen or selected depending where we are looking